src/Docker/Runner/DataLoader/DataLoader.php

```diff
@@ -313,19 +312,7 @@ private function getStagingStorageOutput()
 
     private function validateStagingSetting()
     {
-        if ((($this->getStagingStorageInput() === Reader::STAGING_REDSHIFT) &&
-                ($this->getStagingStorageOutput() !== Reader::STAGING_REDSHIFT)) ||
-            (($this->getStagingStorageOutput() === Reader::STAGING_REDSHIFT) &&
-                ($this->getStagingStorageInput() !== Reader::STAGING_REDSHIFT)) ||
-            (($this->getStagingStorageInput() === Reader::STAGING_SNOWFLAKE) &&
-                ($this->getStagingStorageOutput() !== Reader::STAGING_SNOWFLAKE)) ||
-            (($this->getStagingStorageOutput() === Reader::STAGING_SNOWFLAKE) &&
-                ($this->getStagingStorageInput() !== Reader::STAGING_SNOWFLAKE)) ||
-            (($this->getStagingStorageInput() === Reader::STAGING_SYNAPSE) &&
-                ($this->getStagingStorageOutput() !== Reader::STAGING_SYNAPSE)) ||
-            (($this->getStagingStorageOutput() === Reader::STAGING_SYNAPSE) &&
-                ($this->getStagingStorageInput() !== Reader::STAGING_SYNAPSE))
-        ) {
+        if ($this->getStagingStorageInput() !== $this->getStagingStorageOutput()) {
```

Comment:
that's not the same thing! there's also local which can be combined with anything and s3/abs which can be combined with local

Reply:
ah, right, thank you, i was afraid it was too good to be true :-)

Reply:
hmm, the old conditional doesn't seem to support the claim that `local` can be matched with anything.  I don't see how `local` output and `snowflake` input, for example, doesn't throw this exception.

Reply:
local is not checked at all in the condition - i.e. it doesn't throw the ex

Reply:
ah, I see that now

Reply:
the current condition is not entirely correct either though, because it doesn't allow local+snowflake staging, which is actually allowed. All we need to block is combining different types of workspaces

Reply:
right, that's what confused me.  i'll give it one more go :-)

Reply:
@odinuv ok, i think that 3rd times a charm :-)

Reply:
great, i think you've got it

Reply:
When i run into things like these I feel like i need to go back to school and learn writing conditions :)